### PR TITLE
tcp: make it possible for TCP connections to be creatd by non-TCP pool)

### PR DESCRIPTION
**Pull request by alyssawilk**
As part of #3431 making sure the ALPN pool can create raw TCP active clients by allowing them to be created by a generic connection pool.

Risk Level: low (new pool not used in prod)
Testing: updated tests
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Review by zuercher — APPROVED** [review]

**Comment by alyssawilk** [comment]
ci failure is unrelated: (eventual cause: IOException "No space left on device") so can I get an LGTM on the merge conflict when you get a change?

**Review by ggreenway — APPROVED** [review]

**Pull request merged**
